app/main.py

```diff
@@ -2,0 +1,39 @@
+from typing import List
+
+
+class Animal:
+    alive: List["Animal"] = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self._hidden: bool = False
+        Animal.alive.append(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self._hidden
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, Health: {self.health}, "
+            f"Hidden: {self._hidden}}}"
+        )
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self._hidden = not self._hidden
```

Comment:
Code Efficiency issue: You can change the `bool` value in one line. You can simplify this line like this: `self._hidden = not self._hidden`. Refer to the checklist item 'You can change the `bool` value in one line:' for more details.